include/CLUEstering/CLUEstering.hpp

```diff
@@ -261,6 +311,39 @@ namespace clue {
     alpaka::wait(queue);
   }
 
+  template <uint8_t Ndim>
+  void Clusterer<Ndim>::setupTiles(Queue& queue, const PointsDevice& d_points) {
+    auto nTiles =
+        static_cast<int32_t>(std::ceil(d_points.size() / static_cast<float>(m_pointsPerTile)));
+    const auto nPerDim = static_cast<int32_t>(std::ceil(std::pow(nTiles, 1. / Ndim)));
+    nTiles = static_cast<int32_t>(std::pow(nPerDim, Ndim));
+
+    if (!d_tiles.has_value()) {
+      d_tiles = std::make_optional<TilesDevice>(queue, d_points.size(), nTiles);
+      m_tiles = d_tiles->view();
+    }
+    // check if tiles are large enough for current data
+    if (!(alpaka::trait::GetExtents<clue::device_buffer<Device, int32_t[]>>{}(
+              d_tiles->indexes())[0u] >= d_points.size()) or
+        !(alpaka::trait::GetExtents<clue::device_buffer<Device, int32_t[]>>{}(
+              d_tiles->offsets())[0u] >= static_cast<uint32_t>(nTiles))) {
+      d_tiles->initialize(d_points.size(), nTiles, nPerDim, queue);
+    } else {
+      d_tiles->reset(d_points.size(), nTiles, nPerDim, queue);
+    }
+
+    auto min_max = clue::make_host_buffer<CoordinateExtremes>(queue);
+    auto tile_sizes = clue::make_host_buffer<float[Ndim]>(queue);
+    calculate_tile_size(queue, min_max.data(), tile_sizes.data(), d_points, nPerDim);
+
+    const auto device = alpaka::getDev(queue);
+    alpaka::memcpy(queue, d_tiles->minMax(), min_max);
+    alpaka::memcpy(queue, d_tiles->tileSize(), tile_sizes);
```

Comment:
why you compute them on the device and copy them to the host in `calculate_tile_size`, to copy them again to the device just after it?